hadoop-plugin/src/main/groovy/com/linkedin/gradle/hadoopdsl/job/TensorFlowJob.groovy

```diff
@@ -0,0 +1,244 @@
+  int amCores;
+  int psMemory;
+  int psCores;
+  int workerMemory;
+  int workerCores;
+  int numPs;
+  int numWorkers;
+  String archive;
+  String jar;
+  String taskCommand;
+  Map<String, Object> workerEnv;
+
+  /**
+   * Constructor for a TensorFlowJob.
+   *
+   * @param jobName The job name
+   */
+  TensorFlowJob(String jobName) {
+    super(jobName);
+    workerEnv = new HashMap<>();
+    setJobProperty("type", "TensorFlowJob");
+  }
+
+  /**
+   * Clones the job.
+   *
+   * @return The cloned job
+   */
+  @Override
+  TensorFlowJob clone() {
+    return clone(new TensorFlowJob(name));
+  }
+
+  /**
+   * Helper method to set the properties on a cloned job.
+   *
+   * @param cloneJob The job being cloned
+   * @return The cloned job
+   */
+  TensorFlowJob clone(TensorFlowJob cloneJob) {
+    cloneJob.amMemory = amMemory;
+    cloneJob.amCores = amCores;
+    cloneJob.psMemory = psMemory;
+    cloneJob.psCores = psCores;
+    cloneJob.workerMemory = workerMemory;
+    cloneJob.workerCores = workerCores;
+    cloneJob.numPs = numPs;
+    cloneJob.numWorkers = numWorkers;
+    cloneJob.archive = archive;
+    cloneJob.jar = jar;
+    cloneJob.taskCommand = taskCommand;
+    cloneJob.workerEnv.putAll(workerEnv);
+    return ((TensorFlowJob)super.clone(cloneJob));
+  }
+
+  /**
+   * DSL method to set the am memory.
+   *
+   * @param amMemory amMemory for the TensorFlow job
+   */
+  @HadoopDslMethod
+  void amMemory(int amMemory) {
+    this.amMemory = amMemory;
+    setJobProperty("am_memory", this.amMemory);
+  }
+
+  /**
+   * DSL method to set the am cores.
+   *
+   * @param amCores amCores for the TensorFlow job
+   */
+  @HadoopDslMethod
+  void amCores(int amCores) {
+    this.amCores = amCores;
+    setJobProperty("am_vcores", this.amCores);
+  }
+
+  /**
+   * DSL method to set the ps memory.
+   *
+   * @param psMemory psMemory for the TensorFlow job
+   */
+  @HadoopDslMethod
+  void psMemory(int psMemory) {
+    this.psMemory = psMemory;
+    setJobProperty("ps_memory", this.psMemory);
+  }
+
+  /**
+   * DSL method to set the ps cores.
+   *
+   * @param psCores psCores for the TensorFlow job
+   */
+  @HadoopDslMethod
+  void psCores(int psCores) {
+    this.psCores = psCores;
+    setJobProperty("ps_vcores", this.psCores);
+  }
+
+  /**
+   * DSL method to set the worker memory.
+   *
+   * @param workerMemory workerMemory for the TensorFlow job
+   */
+  @HadoopDslMethod
+  void workerMemory(int workerMemory) {
+    this.workerMemory = workerMemory;
+    setJobProperty("worker_memory", this.workerMemory);
+  }
+
+  /**
+   * DSL method to set the worker cores.
+   *
+   * @param workerCores workerCores for the TensorFlow job
+   */
+  @HadoopDslMethod
+  void workerCores(int workerCores) {
+    this.workerCores = workerCores;
+    setJobProperty("worker_vcores", this.workerCores);
+  }
+
+  /**
+   * DSL method to set the num ps.
+   *
+   * @param numPs numPs for the TensorFlow job
+   */
+  @HadoopDslMethod
```

Comment:
Is this the setting for the number of processes for the job?
Is it the default or the maximum? 
Does it make sense to rename it from **numPs** to **processesCount**?

Reply:
Added some details here.
PS = parameter server (a task type in TensorFlow, similar to a mapper in MR). This is similar to the numExecutors method in Spark DSL.